pkg/brokerapi/fake/server/handler.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+
+	"github.com/pivotal-cf/brokerapi"
+)
+
+// Handler is a fake implementation oif a brokerapi.ServiceBroker
+type Handler struct {
+	Catalog         []brokerapi.Service
+	CatalogRequests int
+
+	ProvisionResp      brokerapi.ProvisionedServiceSpec
+	ProvisionRespError error
+	ProvisionRequests  []ProvisionRequest
+
+	DeprovisionResp     brokerapi.DeprovisionServiceSpec
+	DeprovisonRespErr   error
+	DeprovisionRequests []DeprovisionRequest
+
+	BindResp     brokerapi.Binding
+	BindRespErr  error
+	BindRequests []BindRequest
+
+	UnbindRespErr  error
+	UnbindRequests []UnbindRequest
+
+	UpdateResp     brokerapi.UpdateServiceSpec
+	UpdateRespErr  error
+	UpdateRequests []UpdateRequest
+
+	LastOperationResp     brokerapi.LastOperation
+	LastOperationRespErr  error
+	LastOperationRequests []LastOperationRequest
+}
+
+// NewHandler creates a new fake server handler
+func NewHandler() *Handler {
+	return &Handler{}
+}
+
+// Services is the interface implementation of brokerapi.ServiceBroker
+func (h *Handler) Services(ctx context.Context) []brokerapi.Service {
+	h.CatalogRequests++
+	return h.Catalog
+}
+
+// Provision is the interface implementation of brokerapi.ServiceBroker
+func (h *Handler) Provision(
+	ctx context.Context,
+	instanceID string,
+	details brokerapi.ProvisionDetails,
+	asyncAllowed bool,
+) (brokerapi.ProvisionedServiceSpec, error) {
+	h.ProvisionRequests = append(h.ProvisionRequests, ProvisionRequest{
+		InstanceID:   instanceID,
+		Details:      details,
+		AsyncAllowed: asyncAllowed,
+	})
+	return h.ProvisionResp, h.ProvisionRespError
```

Comment:
I don't see where things like ProvisionResp is defined - I'm guessing a follow-on PR will do that and I'm also guessing that it'll be up to the client/tester to do it. If so,  would it make sense to have some default logic here so that the client/tester doesn't need to hard code all responses, but instead the fake broker will do what's expected in most cases and the client just needs to provide these canned responses if they want something other than the default/happy-path?